overlord/ifacestate/helpers.go

```diff
@@ -95,6 +98,62 @@ func (m *InterfaceManager) addSnaps() error {
 	return nil
 }
 
+func (m *InterfaceManager) unbreakTheWorld() error {
+	// XXX: As a special measure to unbreak the world refresh seccomp security
+	// of all the snaps on the system. Later on this should be improved to
+	// either refresh all security backends or to know how to apply stashed
+	// (versioned) security that was preserved by snapd on upgrade to a new
+	// version of itself.
+
+	// Get all the security backends
+	securityBackends := m.repo.Backends()
+
+	// Get all the snap infos
+	snaps, err := snapstate.ActiveInfos(m.state)
+	if err != nil {
+		return err
+	}
+	// Add implicit slots to all snaps
+	for _, snapInfo := range snaps {
+		snap.AddImplicitSlots(snapInfo)
+	}
+
+	// From now on we don't fail if a particular operation fails, this is a
+	// best-effort service. It's not much of an unbreak-the-world idea if we
+	// bail out and don't start.
+
+	// For each snap:
+	for _, snapInfo := range snaps {
+		snapName := snapInfo.Name()
+		// Get the state of the snap so we can compute the confinement option
+		var snapst snapstate.SnapState
+		if err := snapstate.Get(m.state, snapName, &snapst); err != nil {
+			logger.Noticef("cannot get state of snap %q: %s", snapName, err)
+		}
+
+		// Compute confinement options
+		opts := confinementOptions(snapst.Flags)
+
+		// For each backend:
+		for _, backend := range securityBackends {
+			// The issue this is attempting to fix is only affecting seccomp so
+			// limit the work just to this backend.
+			shouldRefresh := backend.Name() == interfaces.SecuritySecComp
+			if !shouldRefresh {
+				continue
+			}
+			// Refresh security of this snap and backend
+			if err := backend.Setup(snapInfo, opts, m.repo); err != nil {
```

Comment:
How atomic is this operation? Since this happens outside of install/refresh/revert it seems possible for a snap to try to start while this is happening. Is there a race here?